src/coreclr/jit/llvmlower.cpp

```diff
@@ -1129,3 +1140,54 @@ bool Llvm::isInvariantInRange(GenTree* node, GenTree* endExclusive)
 
     return true;
 }
+
+void Llvm::lowerDissolveDependentlyPromotedLocals()
+{
+    // We have now rewritten all references to dependently promoted fields to reference the parent instead.
+    // Drop the annotations such that the subsequent SSA pass will only produce normal (non-composite) names.
+    //
+    for (unsigned lclNum = 0; lclNum < _compiler->lvaCount; lclNum++)
```

Comment:
SSA treats all promoted locals as "sum of their fields", so without this we would get composite SSA definitions that codegen neither understands nor needs.